app/src/main/java/org/groundplatform/android/ui/tos/TermsOfServiceFragment.kt

```diff
@@ -41,13 +67,60 @@ class TermsOfServiceFragment : AbstractFragment() {
     inflater: LayoutInflater,
     container: ViewGroup?,
     savedInstanceState: Bundle?,
-  ): View {
-    val args = TermsOfServiceFragmentArgs.fromBundle(requireArguments())
-    val binding = FragmentTermsServiceBinding.inflate(inflater, container, false)
-    binding.viewModel = viewModel
-    binding.isViewOnly = args.isViewOnly
-    binding.lifecycleOwner = this
-    return binding.root
+  ): View = createComposeView {
+    CreateView(TermsOfServiceFragmentArgs.fromBundle(requireArguments()))
+  }
+
+  @Composable
+  private fun CreateView(args: TermsOfServiceFragmentArgs) {
+    Scaffold(
+      topBar = {
+        Toolbar(
+          stringRes = R.string.tos_title,
+          showNavigationIcon = args.isViewOnly,
```

Comment:
Where is this getting set for the the default cause when the ToS are shown after sign-in (false) and the case when we display the ToS from the app (true)?

Reply:
Toolbar internally has a value true, and this argument has a default value of false

```
 <argument
      android:name="isViewOnly"
      app:argType="boolean" />
```

So, there are two places where this is coming:

```kt
// 1
 MainUiState.TosNotAccepted -> {
        navigateTo(SignInFragmentDirections.showTermsOfService(false))
      }
      
// 2
 R.id.terms_of_service -> {
        findNavController().navigate(HomeScreenFragmentDirections.showTermsOfService(true))
      }
```

Reply:
Ah ok! As long as it's explicitly set somewhere. Thanks!